From 03a638d74d296a17e5e42f23bd014a20cb478eb9 Mon Sep 17 00:00:00 2001
From: Chrissi^ <chris@tinyhost.de>
Date: Wed, 31 Jul 2019 22:36:52 +0200
Subject: [PATCH] gluon-core: Add dependency for KERNEL_NET_ACT_POLICE

This is a fix for the broken ingress traffic-shaping in gluon v2018.2.2
and possibly earlier.

For ingress traffic shaping the kernel option NET_ACT_POLICE is needed.
Before this patch there was no dependency to this. Neither in
gluon_core, gluon-mesh-vpn-core nor in the package.

This patch adds this dependency.

See: https://github.com/freifunk-gluon/gluon/issues/1790
---
 package/gluon-core/Config.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/package/gluon-core/Config.in b/package/gluon-core/Config.in
index caf4f4c6c..1da7f2947 100644
--- a/package/gluon-core/Config.in
+++ b/package/gluon-core/Config.in
@@ -50,6 +50,10 @@ config KERNEL_NET_CLS_ACT
 	bool
 	select KERNEL_NET_CLS
 
+config KERNEL_NET_ACT_POLICE
+	bool
+	select KERNEL_NET_CLS_ACT
+
 config KERNEL_NET_CLS_BASIC
 	bool
 	select KERNEL_NET_CLS
@@ -343,3 +347,4 @@ config GLUON_SPECIALIZE_KERNEL
 	select KERNEL_IP_NF_NAT
 	select KERNEL_IP_NF_TARGET_MASQUERADE
 	select KERNEL_IP_NF_TARGET_REDIRECT
+	select KERNEL_NET_ACT_POLICE
